### ISSUE TITLE
org-agenda breaks instead of displaying todos

### ISSUE THREAD

**Issue by countkrusty** [syl20bnr/spacemacs#13386]
Description :octocat:

Listing all TODO entries using org-agenda throws an error: wrong-number-of-arguments (0 . 0) 2
I already tried to set the agenda files in my user-config as suggested in #7305 , but that didn't change anything

Reproduction guide 🐞

- Start Emacs
- open org-agenda using SPC-a-o-o
- press t for a list of all TODO entries

Observed behaviour: 👀 💔

Trying to open all todo entries or even the weekly agenda results in an error while agenda buffer remains empty

Expected behaviour: ❤️ 😄

org-agenda should display a list of all todo entries in the files added to org-agenda-files

System Info 💻

- OS: darwin
- Emacs: 26.3
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 2049817b1)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(html helm auto-completion emacs-lisp org git
      (python :variables python-test-runner 'pytest python-enable-yapf-format-on-save t)
      gnuplot dotspacemacs-additional-packages
      '(‘org-indent-initialize-agent))
```
- System configuration features: NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS NS MODULES THREADS

Backtrace 🐾

```
  Debugger entered--Lisp error: (wrong-number-of-arguments (0 . 0) 2)
  org-get-tags(nil nil)
  org-agenda-get-todos()
  org-agenda-get-day-entries("~/Documents/org/todo.org" (3 16 2020) :todo)
  org-todo-list(nil)
  funcall-interactively(org-todo-list nil)
  call-interactively(org-todo-list)
  org-agenda(nil)
  funcall-interactively(org-agenda nil)
  call-interactively(org-agenda nil nil)
  command-execute(org-agenda)
```

**Comment by duianto** [comment]
It seems to be working as expected for me, with the following org layer setting:
```elisp
     (org :variables org-agenda-files '("~/org/notes.org"))
```
`SPC a o o t` shows the todo entries in the `notes.org` file.
`SPC a o o a` shows the weeks scheduled entries.

In both of the following OSs, on the latest `develop` branch with updated packages.

<details><summary>
PopOS 19.10
</summary>
<pre>
#### System Info :computer:
- OS: gnu/linux
- Emacs: 26.3.50
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. d7868dae2)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(auto-completion command-log common-lisp emacs-lisp git helm markdown multiple-cursors
                 (org :variables org-agenda-files
                      '("~/org/notes.org"))
                 python
                 (shell :variables shell-default-shell 'term shell-default-height 30 shell-default-position 'bottom)
                 spell-checking syntax-checking treemacs version-control)
```
- System configuration features: XPM JPEG TIFF GIF PNG SOUND DBUS GSETTINGS GLIB NOTIFY LIBSELINUX GNUTLS FREETYPE XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM MODULES THREADS

</pre>
</details>
<details><summary>
Windows 1903
</summary>
<pre>
#### System Info :computer:
- OS: windows-nt
- Emacs: 26.3
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. d7868dae2)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(autohotkey clojure command-log common-lisp emacs-lisp ess git helm imenu-list javascript
            (markdown :variables markdown-live-preview-engine 'vmd markdown-command "vmd")
            multiple-cursors
            (org :variables org-agenda-files
                 '("~/org/notes.org"))
            pdf python
            (shell :variables shell-default-shell 'shell shell-default-height 30 shell-default-position 'bottom)
            spell-checking syntax-checking treemacs version-control)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG SOUND NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS THREADS LCMS2

</pre>
</details>

**Comment by countkrusty** [comment]
Thanks for taking a look at this!
Unfortunately, no matter where I define the org-agenda-files variable, the error doesn't change and todos are not displayed. I tried to define it 1) in the layers directly, like you did, 2) with-eval-after-load in  `user-config` and, using `C-c-[ ` in the file directly, 3) in the `custom-set-variables`. I checked for all of these options that the variable is correctly read by putting in a non-existing file, which throws the correct error. I'm a spacemacs/org-mode newbie so I'm unsure what else I could try. My .spacemacs itself is written in org-mode (To set this up, I followed the documentation here: (https://orgmode.org/worg/org-contrib/babel/intro.html). Could this be part of the error? Apart from org-agenda everything else is running fine.

**Comment by duianto** [comment]
Make sure that your packages are up to date, you can click on the `[Update Packages]` button on the Spacemacs home buffer `SPC b h`.

My packages are up to date, and `M-x org-version` shows:
`Org mode version 9.3.6 (9.3.6-25-g685b2c-elpaplus @ c:/Users/username/.emacs.d/elpa/26.3/develop/org-plus-contrib-20200323/)`

If that doesn't work then here's a thread where the same error message is discussed:
https://stackoverflow.com/questions/54580647/org-agenda-wrong-number-of-arguments-error

The first two answers talks about downgrading org, I'm not completely sure how to do that, but the third answer says:
>I had this problem on a new install of spacemacs in mac osx. Recompiling elpa fixed it:
```elisp
M-: (byte-recompile-directory package-user-dir nil 'force)
```

**Comment by countkrusty** [comment]
Thanks for all your suggestions! Updating the packages worked only transiently, i.e. when I restarted Spacemacs I ended up with an older version of org-mode again. I know one can find a lot of issues about having a .spacemacs.org file causes conflicts with different versions of org and so on. Since I was too lazy to get into where I went wrong I wiped and re-installed spacemacs and customized the .spacemacs file directly with elisp syntax only. Thanks again for all your help!
